scala-mode-syntax.el

```diff
@@ -155,6 +155,29 @@
           "\\|" scala-syntax:symbolLiteral-re
           "\\|" "null" "\\)"))
 
+(defconst scala-syntax:interpolation-re
+  (concat "\\(" "\\$"  scala-syntax:id-re "\\|" "\\${[^}\n\\\\]*}" "\\)"))
```

Comment:
in general this lokos good. However, the `[^}]*` breaks for nested interpolation.

```scala
s"foo ${s"bar ${s"$zot"} qux"} quux"
```

would be highlighten from the first `$` until the first `}`, so `qux }` would not be highlighted.

While I think this is a theoretical case -- and I'm willing to accept the pull request as is, there would be a simple way around this by using the parser to jump over balanced parens, if one would be willing to attempt using it.